platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ProblemReportingCrossProjectModelAccess.kt

```diff
@@ -1037,9 +1040,70 @@ class ProblemReportingCrossProjectModelAccess(
             problems.onProblem(problem)
             coupledProjectsListener.onProjectReference(referrer.owner, delegate.owner)
             // Configure the target project, if it would normally be configured before the referring project
-            if (delegate.compareTo(referrer) < 0 && delegate.parent != null) {
+            if (delegate < referrer && delegate.parent != null && buildModelParameters.isInvalidateCoupledProjects) {
                 delegate.owner.ensureConfigured()
             }
         }
+
+        private
+        fun <T> DefaultProject.withEvaluationOrderCheck(action: DefaultProject.() -> T?): T? {
+            // Attempt to get a result, since plugin might to contribute to DynamicObject properties
+            var failure: Throwable? = null
+            val result =
```

Comment:
Maybe there is a some utility function for such scenarios?

Reply:
There is a function in Kotlin stdlib that might be helpful:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/run-catching.html

Reply:
@alllex Thanks! Forgot about it